internal/context/context.go

```diff
@@ -0,0 +1,54 @@
+package context
+
+import (
+	stdcontext "context"
+)
+
+type (
+	// Context augments the context.Context interface with addition methods
+	// to allow to retrieve the extra attributes our wrapper context holds
+	Context interface {
+		stdcontext.Context
+		Method() string
+		Arguments() map[string]interface{}
```

Comment:
I'm pretty out of the loop, and I'm not very familiar with the best practices for the context package, so please excuse me if I'm asking some stupid questions here.

What's the reason for extending the context package with a custom context type, instead of say just storing the method, arguments, and correlation ID via the existing Values/Value feature of the context package? Or is it more common to create a custom type that adds support for the custom stuff you need?

Reply:
Just to reduce the amount of casting required as this will be in all hot paths of our code. In addition to this having our own context allows us to do custom logic for it in the logger.

Reply:
By casting are you referring to the fact that each call to `WithValue()` creates a new context from a parent context with a single value added?